dscreator/sources/ferrybox/extractor.py

```diff
@@ -145,19 +148,58 @@ def fetch_slice(
 
         The all list should have the same length and time should be increasing. Missing value returned as 9 for qc
         """
-        uuids, wls = zip(*(self.variable_uuid_map[vcode].split("_") for vcode in self.variable_codes))
-        data_list = queries.get_spectra(
-            self.engine,
-            track_uuid=self.variable_uuid_map["track"],
-            uuids=list(uuids),
-            wave_lengths=list(wls),
-            start_time=start_time,
-            end_time=end_time,
-            qc_flags=self.qc_flags,
-        )
+        # Separate rrs and other radiance/irradiance variables
+        rrs_codes = [vcode for vcode in self.variable_codes if vcode.startswith("rrs_")]
+        radiancelu_codes = [vcode for vcode in self.variable_codes if vcode.startswith("radiancelu_")]
+        radianceld_codes = [vcode for vcode in self.variable_codes if vcode.startswith("radianceld_")]
+        irradianceed_codes = [vcode for vcode in self.variable_codes if vcode.startswith("irradianceed_")]
+
+        data_list = []
+
+        # Query rrs with QC flag 1 - this determines which datapoints we keep
+        if rrs_codes:
+            rrs_uuids, rrs_wls = zip(*(self.variable_uuid_map[vcode].split("_") for vcode in rrs_codes))
+            rrs_data = queries.get_spectra(
+                self.engine,
+                track_uuid=self.variable_uuid_map["track"],
+                uuids=list(rrs_uuids),
+                wave_lengths=list(rrs_wls),
+                start_time=start_time,
+                end_time=end_time,
+                qc_flags=[1],
+            )
+            data_list.extend(rrs_data)
+
+            # Build set of (time, wavelength) tuples where rrs has QC=1
+            valid_time_wl_pairs = {(row["time"], row["uuid"].split("_")[-1]) for row in rrs_data}
+        else:
+            valid_time_wl_pairs = set()
+
+        # Query other variables (radiancelu, radianceld, irradianceed) with all QC flags
+        # but only keep matching (time, wavelength) pairs where rrs has QC=1
+        for var_codes in [radiancelu_codes, radianceld_codes, irradianceed_codes]:
+            if var_codes and valid_time_wl_pairs:
+                var_uuids, var_wls = zip(*(self.variable_uuid_map[vcode].split("_") for vcode in var_codes))
+                var_data = queries.get_spectra(
+                    self.engine,
+                    track_uuid=self.variable_uuid_map["track"],
+                    uuids=list(var_uuids),
+                    wave_lengths=list(var_wls),
+                    start_time=start_time,
+                    end_time=end_time,
+                    qc_flags=[0, 1, -1],  # Accept all QC flags
+                )
+                # Filter to only include data where corresponding rrs had QC=1
+                filtered_var_data = [
+                    row for row in var_data if (row["time"], row["uuid"].split("_")[-1]) in valid_time_wl_pairs
+                ]
+                data_list.extend(filtered_var_data)
```

Comment:
I had perhaps looked at updating the query to return  radiancelu, radianceld and irradianceed for a given time where rrs=1 ordered by time, in my head that is more efficient, but don't know if it is possible